Projects/ChangeTier/Sources/Views/CompareTierScreen.swift

```diff
@@ -45,8 +66,14 @@ struct CompareTierScreen: View {
 }
 
 extension ChangeTierViewModel {
-    fileprivate func getFilteredPerils(currentTier: Tier) -> [Perils] {
-        var currentPerils = currentTier.quotes.first?.productVariant?.perils ?? []
+    fileprivate func getFilteredPerils(currentTier: Tier, selectedQuoteId: String?) -> [Perils] {
+        var currentPerils =
+            currentTier.quotes.first { quote in
+                quote.id == selectedQuoteId
+            }?
+            .productVariant?
+            .perils ?? currentTier.quotes.first?.productVariant?.perils ?? []
+
         let otherPerils = self.tiers.filter({ $0.id != currentTier.id })
             .reduce(into: [Perils]()) { partialResult, tier in
                 return partialResult.append(contentsOf: tier.quotes.first?.productVariant?.perils ?? [])
```

Comment:
can we also try to match quotes if possible instead of taking the first one?
tier.quotes.first?.productVariant?.perils ?? []